### PR TITLE
Track import location changes

### PR DESCRIPTION
**Pull request by jsiirola**
The LP writer rewrite will propose that we move certain Pyomo utilities to more centralized / less obscure locations.  This PR prepares this library for that change.

This also fixes a bug in the performance tester where test markers put on the test function were not being picked up by the test driver.

### CODE REVIEW

**Comment by mrmundt** [comment]
Love it. I am assuming we should wait to merge this until the new LP writer goes in, yes?

File: cute/test_CUTE.py [hunk 1]
```diff
@@ -26,8 +26,17 @@
...
+    _loaded = False
+if not _loaded:
+    try:
+        from pyomo.repn.tests.ampl.nl_diff import load_and_compare_nl_baseline
+        _loaded = True
+    except ImportError:
+        _loaded = False
+if not _loaded:
     # Backwards compatibility: fallback to the previous differ
     # TODO: This can be removed after the NL Writer v2 has been merged
```
Comment (mrmundt) [inline cute/test_CUTE.py:41]: Can we maybe do this now since we merged NL V2?

Comment (jsiirola) [reply cute/test_CUTE.py:41]: I am not sure.  If we managed these two repos as a perfectly matched set, then yes.  But leaving it in lets us run the current HEAD here against older Pyomo releases (good for benchmarking / comparisons)

**Comment by jsiirola** [comment]
> Love it. I am assuming we should wait to merge this until the new LP writer goes in, yes?

This should be safe to merge any time (and before the LP writer might let me remove some backwards compatibility shims that I needed to get testing running).

**Review by mrmundt — APPROVED** [review]

**Pull request merged**